src/ui/shared/requests/useQuotes.ts

```diff
@@ -1,46 +1,156 @@
-import { useCallback, useMemo, useState } from 'react';
-import omit from 'lodash/omit';
 import type { EmptyAddressPosition } from '@zeriontech/transactions';
-import { commonToBase } from 'src/shared/units/convert';
-import {
-  getAssetImplementationInChain,
-  getDecimals,
-} from 'src/modules/networks/asset';
+import { getAssetImplementationInChain } from 'src/modules/networks/asset';
 import { createChain } from 'src/modules/networks/Chain';
-import { DEFI_SDK_TRANSACTIONS_API_URL } from 'src/env/config';
 import { isNumeric } from 'src/shared/isNumeric';
-import type { Quote, TransactionDescription } from 'src/shared/types/Quote';
-import { createUrl } from 'src/shared/createUrl';
 import type { AddressPosition } from 'defi-sdk';
 import { getSlippageOptions } from 'src/ui/pages/SwapForm/SlippageSettings/getSlippageOptions';
+import { useCallback, useMemo, useState } from 'react';
+import type { Quote } from 'src/shared/types/Quote';
+import { getBaseQuantity } from 'src/modules/networks/asset';
+import type { Chain } from 'src/modules/networks/Chain';
+import { DEFI_SDK_TRANSACTIONS_API_URL } from 'src/env/config';
+import { createUrl } from 'src/shared/createUrl';
+import omit from 'lodash/omit';
 import { useEventSource } from './useEventSource';
 
+type QuoteSortType = 'amount' | 'time';
+
 interface QuotesParams {
+  inputAssetId: string;
+  outputAssetId: string;
+  inputChain: Chain;
+  outputChain: Chain | null;
+  inputAmount?: string;
+  outputAmount?: string;
+  from?: string;
+  slippage?: string;
+  gasPrice?: string;
+  priorityFee?: string;
+  maxFee?: string;
+  sourceId?: string;
+  sort?: QuoteSortType;
+  refetchHash?: number;
+  to?: string;
+}
+
+function getQuotesSearchParams(params: QuotesParams): URLSearchParams {
+  const searchParams = new URLSearchParams({
+    input_asset_id: params.inputAssetId,
+    output_asset_id: params.outputAssetId,
```

Comment:
`input_token` and `output_token` parameters are deprecated and replaced with those having the `_asset_id` suffix